### ISSUE TITLE
Abandoned OAuth flow blocks session requests

### ISSUE THREAD

**Issue by rtandy** [aws-amplify/amplify-js#12814]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication

### Amplify Version

v6

### Amplify Categories

auth

### Backend

CDK

### Environment information

<details>

React v18.2.0
Amplify Auth v6.0.10

Further details available upon request

</details>

### Describe the bug

Hello, thank you for your work on Amplify.

I am migrating a React app from [amazon-cognito-identity-js](https://www.npmjs.com/package/amazon-cognito-identity-js) to Amplify Auth v6 with Cognito Hosted UI. I am not using any other Amplify features at this time.

On load, I call `fetchAuthSession` to check for a session, and `signInWithRedirect` if needed.

With Amplify Auth v6.0.10, after the user is redirected to login, if they close the Hosted UI tab without signing in, then the next time they navigate to the application, `fetchAuthSession` never resolves and the app hangs waiting for a session.

It looks like the OAuth in-flight key is being set in local storage, but when the user closed the sign-in tab, nothing ever cleared it, so the next time we navigate to the app, `fetchAuthSession` is blocked forever.

I'm not sure if I should be calling something to clear/reset the OAuth state before trying `fetchAuthSession`?

With Amplify Auth 6.0.9, everything works. The issue is new in 6.0.10.

By the way, is the type `AuthSession` (return type of fetchAuthSession) exported anywhere? I checked `aws-amplify`, `aws-amplify/auth`, `@aws-amplify/core`, and `@aws-amplify/auth`. (I'm importing `fetchAuthSession` from `aws-amplify/auth`.)

Please let me know if you require any further information. Thank you!

### Expected behavior

User abandoning an OAuth flow should not cause `fetchAuthSession` to be blocked forever.

### Reproduction steps

1. User navigates to React application
2. Application redirects to OAuth login
3. User abandons OAuth flow (closes browser tab)
4. User navigates to application again

### Code Snippet

```typescript
import { useEffect, useState } from 'react'
import { Amplify } from 'aws-amplify'
import { JWT, fetchAuthSession, signInWithRedirect, signOut } from 'aws-amplify/auth'

type AuthState = 'loading' | 'signed-out' | 'signed-in'

Amplify.configure({
    Auth: {
        Cognito: {
            userPoolId: import.meta.env.VITE_USER_POOL_ID,
            userPoolClientId: import.meta.env.VITE_CLIENT_ID,
            loginWith: {
                oauth: {
                    domain: import.meta.env.VITE_OAUTH_DOMAIN,
                    scopes: ['openid', 'email', 'profile'],
                    redirectSignIn: ['http://localhost:5173'],
                    redirectSignOut: ['http://localhost:5173'],
                    responseType: 'code'
                }
            }
        }
    }
})

export default function App() {
    const [authState, setAuthState] = useState<AuthState>('loading')
    const [idToken, setIdToken] = useState<JWT | null>(null)

    useEffect(() => {
        (async () => {
            try {
                console.log('Calling fetchAuthSession()')
                const session = await fetchAuthSession()
                console.log('fetchAuthSession result:', session)
                if (session.tokens && session.tokens.idToken) {
                    setAuthState('signed-in')
                    setIdToken(session.tokens.idToken)
                } else {
                    setAuthState('signed-out')
                }
            } catch (err) {
                console.error(err)
                setAuthState('signed-out')
            }
        })()
    }, [])

    switch (authState) {
        case 'loading':
            return <>Loading...</>
        case 'signed-out':
            signInWithRedirect()
            return <>Redirecting to sign-in page...</>
        case 'signed-in':
            const email = idToken!.payload.email as string
            return <>
                <p>Welcome, {email}!</p>
                <button onClick={() => signOut()}>Sign out</button>
            </>
    }
}
```

**Comment by cwomack** [comment]
Hey, @rtandy 👋.  We're looking into this now and will let you know if there are further questions.

**Comment by cwomack** [comment]
@rtandy, can you confirm that a refresh of the browser after navigating back to the application resolves the issue for you?

**Comment by rtandy** [comment]
Yes, it does.

**Comment by cwomack** [comment]
Appreciate the confirmation, @rtandy.  We have a fix in progress already and will update this issue when it's released.

**Comment by rtandy** [comment]
Thank you!

**Comment by HuiSF** [comment]
Hi @rtandy I made a patch version including the fix in the above linked PR and the fix worked well in my testing, would you mind to give it a try as well? You can get the patch version by

```sh
npm install aws-amplify@6.0.11-oauth-patch.157b3d4.0
```

**Comment by cwomack** [comment]
The fix for this is now out! With @rtandy's confirmation and fix live in v 6.0.11, we'll close this issue.
